tests/unit/lib/functions/runtimes/go/index.test.js

```diff
@@ -0,0 +1,29 @@
+const test = require('ava')
+const sinon = require('sinon')
+
+const { rewiremock } = require('../../../../../integration/utils/rewiremock')
+
+const runFunctionsProxySpy = sinon.stub()
+// eslint-disable-next-line n/global-require
+const { invokeFunction } = rewiremock.proxy(() => require('../../../../../../src/lib/functions/runtimes/go/index'), {
+  '../../../../../../src/lib/functions/local-proxy': {
+    runFunctionsProxy: runFunctionsProxySpy,
+  },
+})
+
+const invokeFunctionMacro = test.macro({
+  async exec(t, prop, expected) {
+    runFunctionsProxySpy.returns(Promise.resolve({ stdout: JSON.stringify({ [prop]: expected }) }))
```

Comment:
[dust] I think you could simplify this a bit by using `resolves()`.

```suggestion
    runFunctionsProxySpy.resolves({ stdout: JSON.stringify({ [prop]: expected }) })
```

Reply:
I do this in the upcoming rust PR